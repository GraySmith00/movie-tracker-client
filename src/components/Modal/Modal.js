import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { addTrailerToState } from '../../actions/movieActions';

export class Modal extends Component {
  componentDidUpdate() {
    this.modalTarget = document.createElement('div');
    this.modalTarget.className = 'modal';
    document.body.appendChild(this.modalTarget);
    this._render();
  }

  handleClick = () => {
    ReactDOM.unmountComponentAtNode(this.modalTarget);
    document.body.removeChild(this.modalTarget);
  };

  _render() {
    if (this.props.trailer) {
      ReactDOM.render(
        <div>
          <iframe
            className="modal-trailer"
            title="modal-trailer"
            src={this.props.trailer}
            width="800"
            height="500"
          />
          <button className="remove-modal" onClick={this.handleClick}>
            X
          </button>
        </div>,
        this.modalTarget
      );
    }
  }

  render() {
    return <noscript />;
  }
}

const mapStateToProps = state => ({
  trailer: state.movies.trailer
});

const mapDispatchToProps = dispatch => ({
  addTrailerToState: trailer => dispatch(addTrailerToState(trailer))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Modal);
