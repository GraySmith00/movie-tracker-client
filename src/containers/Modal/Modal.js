import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';

export class Modal extends Component {
  componentDidUpdate() {
    this.modalTarget = document.createElement('div');
    this.modalTarget.className = 'modal';
    document.body.appendChild(this.modalTarget);
    this._render();
  }

  handleClick = () => {
    ReactDOM.unmountComponentAtNode(this.modalTarget);
    document.body.removeChild(this.modalTarget);
  };

  _render() {
    if (this.props.trailer) {
      ReactDOM.render(
        <div>
          <iframe
            className="modal-trailer"
            title="modal-trailer"
            src={this.props.trailer}
            width="800"
            height="500"
          />
          <button className="remove-modal" onClick={this.handleClick}>
            X
          </button>
        </div>,
        this.modalTarget
      );
    }
  }

  render() {
    return <noscript />;
  }
}

Modal.propTypes = {
  trailer: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  trailer: state.movies.trailer
});

export default connect(mapStateToProps)(Modal);
