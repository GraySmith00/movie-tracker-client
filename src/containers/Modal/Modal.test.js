import React from 'react';
import { shallow } from 'enzyme';

import { Modal } from './Modal';

describe('Modal component', () => {
  let wrapper;
  let mockTrailer;

  beforeEach(() => {
    mockTrailer = 'https://www.youtube.com/embed/bsLk0NPRFAc?autoplay=1';
    wrapper = shallow(<Modal trailer={mockTrailer} />);
  });

  it('should match the snapshot with a trailer', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match the snapshot with no trailer', () => {
    wrapper = shallow(<Modal trailer={mockTrailer} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
