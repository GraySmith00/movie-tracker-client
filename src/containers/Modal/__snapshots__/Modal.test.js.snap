// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should match the snapshot with a trailer 1`] = `"<noscript></noscript>"`;

exports[`Modal component should match the snapshot with no trailer 1`] = `"<noscript></noscript>"`;
